### PR TITLE
feat: Allow explicitly setting consumerVersionSelectors through PactB…

### PR DESCRIPTION
**Pull request by pendsley**
…roker annotation.

### CODE REVIEW

File: provider/src/main/java/au/com/dius/pact/provider/junitsupport/loader/VersionSelector.java [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+ * Used to specify which versions to use when querying the Pact matrix.
+ */
+@Retention(RetentionPolicy.RUNTIME)
+@Target(ElementType.TYPE)
+@Inherited
+public @interface VersionSelector {
+    /**
+     * @return Tags to use to fetch pacts for, defaults to `latest`
+     */
+    String tag() default "latest";
```
Comment (uglyog) [inline provider/src/main/java/au/com/dius/pact/provider/junitsupport/loader/VersionSelector.java:19]: I don't think latest here is right. It probably should be empty or mandatory.

**Comment by uglyog** [comment]
Build is failing due to lint errors:

```
/home/travis/build/DiUS/pact-jvm/provider/src/main/kotlin/au/com/dius/pact/provider/junitsupport/loader/PactBrokerLoader.kt:33:9: Lint error > [parameter-list-wrapping] Unexpected indentation (expected 2, actual 8)
/home/travis/build/DiUS/pact-jvm/provider/src/main/kotlin/au/com/dius/pact/provider/junitsupport/loader/PactBrokerLoader.kt:34:9: Lint error > [parameter-list-wrapping] Unexpected indentation (expected 2, actual 8)
/home/travis/build/DiUS/pact-jvm/provider/src/main/kotlin/au/com/dius/pact/provider/junitsupport/loader/PactBrokerLoader.kt:35:9: Lint error > [parameter-list-wrapping] Unexpected indentation (expected 2, actual 8)
/home/travis/build/DiUS/pact-jvm/provider/src/main/kotlin/au/com/dius/pact/provider/junitsupport/loader/PactBrokerLoader.kt:36:9: Lint error > [parameter-list-wrapping] Unexpected indentation (expected 2, actual 8)
/home/travis/build/DiUS/pact-jvm/provider/src/main/kotlin/au/com/dius/pact/provider/junitsupport/loader/PactBrokerLoader.kt:39:9: Lint error > [parameter-list-wrapping] Unexpected indentation (expected 2, actual 8)
/home/travis/build/DiUS/pact-jvm/provider/src/main/kotlin/au/com/dius/pact/provider/junitsupport/loader/PactBrokerLoader.kt:40:9: Lint error > [parameter-list-wrapping] Unexpected indentation (expected 2, actual 8)
/home/travis/build/DiUS/pact-jvm/provider/src/main/kotlin/au/com/dius/pact/provider/junitsupport/loader/PactBrokerLoader.kt:41:9: Lint error > [parameter-list-wrapping] Unexpected indentation (expected 2, actual 8)
/home/travis/build/DiUS/pact-jvm/provider/src/main/kotlin/au/com/dius/pact/provider/junitsupport/loader/PactBrokerLoader.kt:42:9: Lint error > [parameter-list-wrapping] Unexpected indentation (expected 2, actual 8)
/home/travis/build/DiUS/pact-jvm/provider/src/main/kotlin/au/com/dius/pact/provider/junitsupport/loader/PactBrokerLoader.kt:43:9: Lint error > [parameter-list-wrapping] Unexpected indentation (expected 2, actual 8)
/home/travis/build/DiUS/pact-jvm/provider/src/main/kotlin/au/com/dius/pact/provider/junitsupport/loader/PactBrokerLoader.kt:44:9: Lint error > [parameter-list-wrapping] Unexpected indentation (expected 2, actual 8)
/home/travis/build/DiUS/pact-jvm/provider/src/main/kotlin/au/com/dius/pact/provider/junitsupport/loader/PactBrokerLoader.kt:45:9: Lint error > [parameter-list-wrapping] Unexpected indentation (expected 2, actual 8)
/home/travis/build/DiUS/pact-jvm/provider/src/main/kotlin/au/com/dius/pact/provider/junitsupport/loader/PactBrokerLoader.kt:46:9: Lint error > [parameter-list-wrapping] Unexpected indentation (expected 2, actual 8)
/home/travis/build/DiUS/pact-jvm/provider/src/main/kotlin/au/com/dius/pact/provider/junitsupport/loader/PactBrokerLoader.kt:113:76: Lint error > [curly-spacing] Unexpected space after "}"
/home/travis/build/DiUS/pact-jvm/provider/src/main/kotlin/au/com/dius/pact/provider/junitsupport/loader/PactBrokerLoader.kt:113:77: Lint error > [dot-spacing] Unexpected spacing before "."
```

File: provider/src/main/java/au/com/dius/pact/provider/junitsupport/loader/VersionSelector.java (hunk 1 above)
Comment (pendsley) [reply provider/src/main/java/au/com/dius/pact/provider/junitsupport/loader/VersionSelector.java]: Removed the default and made it mandatory.

**Comment by pendsley** [comment]
Looks like the travis build is failing because it timed out trying to download gradle.

> $ ./gradlew --version
> Downloading https://services.gradle.org/distributions/gradle-6.3-all.zip
> Exception in thread "main" java.net.ConnectException: Connection timed out (Connection timed out)
> The command "./gradlew --version" exited with 1.
> 

But the overall build still succeeded

> BUILD SUCCESSFUL in 2m 43s
> 26 actionable tasks: 26 executed
> The command "./gradlew --stacktrace --no-daemon $PROJECT build" exited with 0.

**Comment by anto-ac** [comment]
I've re-triggered the failed build configuration

**Comment by uglyog** [comment]
Thanks for the changes! Good to merge.

**Comment by pendsley** [comment]
Closes #1142

**Pull request merged**